sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/ORCCarbonWriter.java

```diff
@@ -0,0 +1,168 @@
+    ObjectInspector objectInspector = orcReader.getObjectInspector();
+    RecordReader recordReader = orcReader.rows();
+    if (objectInspector instanceof StructObjectInspector) {
+      StructObjectInspector structObjectInspector =
+          (StructObjectInspector) orcReader.getObjectInspector();
+      while (recordReader.hasNext()) {
+        Object record = recordReader.next(null); // to remove duplicacy.
+        List valueList = structObjectInspector.getStructFieldsDataAsList(record);
+        for (int i = 0; i < valueList.size(); i++) {
+          valueList.set(i, parseOrcObject(valueList.get(i), 0));
+        }
+        this.csvCarbonWriter.write(valueList.toArray());
+      }
+    } else {
+      while (recordReader.hasNext()) {
+        Object record = recordReader.next(null); // to remove duplicacy.
+        this.csvCarbonWriter.write(new Object[]{parseOrcObject(record, 0)});
+      }
+    }
+  }
+
+  private String parseOrcObject(Object recordObject, int level) {
+    if (recordObject instanceof OrcStruct) {
+      Objects.requireNonNull(orcReader);
+      StructObjectInspector structObjectInspector = (StructObjectInspector) orcReader
+          .getObjectInspector();
+      List value = structObjectInspector.getStructFieldsDataAsList(recordObject);
+      for (int i = 0; i < value.size(); i++) {
+        value.set(i, parseOrcObject(value.get(i), level + 1));
+      }
+      String str = listToString(value, level);
+      if (str.length() > 0) {
+        return str.substring(0, str.length() - 1);
+      }
+      return null;
+    } else if (recordObject instanceof ArrayList) {
+      ArrayList listValue = (ArrayList) recordObject;
+      for (int i = 0; i < listValue.size(); i++) {
+        listValue.set(i, parseOrcObject(listValue.get(i), level + 1));
+      }
+      String str = listToString(listValue, level);
+      if (str.length() > 0) {
+        return str.substring(0, str.length() - 1);
+      }
+      return null;
+    } else if (recordObject instanceof LinkedHashMap) {
+      LinkedHashMap<Text, Object> keyValueRow = (LinkedHashMap<Text, Object>) recordObject;
+      for (Map.Entry<Text, Object> entry : keyValueRow.entrySet()) {
+        Object val = parseOrcObject(keyValueRow.get(entry.getKey()), level + 2);
+        keyValueRow.put(entry.getKey(), val);
+      }
+      StringBuilder str = new StringBuilder();
+      for (Map.Entry<Text, Object> entry : keyValueRow.entrySet()) {
+        Text key = entry.getKey();
+        str.append(key.toString()).append("$").append(keyValueRow.get(key)).append("#");
+      }
+      if (str.length() > 0) {
+        return str.substring(0, str.length() - 1);
+      }
+      return null;
+    }
+    if (recordObject == null) {
+      return null;
+    }
+    return recordObject.toString();
+  }
+
+  private String listToString(List value, int level) {
+    String delimiter = "";
+    if (level == 0) {
+      delimiter = "#";
+    }
+    else if (level == 1) {
+      delimiter = "$";
```

Comment:
It's  better to support configure it by user

Reply:
This is just for internal implementation. Not required to provide support to configure by user.